src/battle_ai_main.c

```diff
@@ -2673,7 +2674,11 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 ADJUST_SCORE(-10);
             break;
         case EFFECT_UPPER_HAND:
-            if (predictedMove == MOVE_NONE || IsBattleMoveStatus(predictedMove) || AI_IsSlower(battlerAtk, battlerDef, move) || GetBattleMovePriority(battlerDef, predictedMove) < 1 || GetBattleMovePriority(battlerDef, predictedMove) > 3) // Opponent going first or not using priority move
+            if (predictedMove == MOVE_NONE
+             || IsBattleMoveStatus(predictedMove)
+             || AI_IsSlower(battlerAtk, battlerDef, move)
+             || GetBattleMovePriority(battlerDef, aiData->abilities[battlerDef], predictedMove) < 1
+             || GetBattleMovePriority(battlerDef, aiData->abilities[battlerAtk], predictedMove) > 3) // Opponent going first or not using priority move
```

Comment:
Wouldn't it be faster to only call this function once and store it in a variable?